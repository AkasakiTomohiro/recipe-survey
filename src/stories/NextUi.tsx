import React from 'react';
import { Button } from '@nextui-org/react';

interface NextUiProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * Button contents
   */
  label: string;
  isLoading?: boolean;
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

/**
 * Primary UI component for user interaction
 */
export const NextUi = ({
  ...props
}: NextUiProps) => {
  return (
    <Button color={props.primary ? 'primary' : 'secondary'} isLoading={props.isLoading ? props.isLoading : false}>
      {props.label}
    </Button>
  );
};
