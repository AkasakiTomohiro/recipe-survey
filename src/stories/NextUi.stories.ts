import type { Meta, StoryObj } from '@storybook/react';

import { NextUi } from './NextUi';

const meta = {
  title     : 'Example/NextUi',
  component : NextUi,
  parameters: {
    layout: 'centered'
  },
  tags    : ['autodocs'],
  argTypes: {
    backgroundColor: { control: 'color' }
  }
} satisfies Meta<typeof NextUi>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    primary: true,
    label  : 'Button'
  }
};

export const Secondary: Story = {
  args: {
    label    : 'Loading',
    isLoading: true,
    size     : 'md'
  }
};
