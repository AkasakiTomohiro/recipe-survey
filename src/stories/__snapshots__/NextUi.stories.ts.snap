// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/NextUi Primary smoke-test 1`] = `
<div data-overlay-container="true">
  <button class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-6 min-w-unit-24 h-unit-12 text-medium gap-unit-3 rounded-large [&amp;>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
          type="button"
  >
    Button
  </button>
</div>
`;

exports[`Example/NextUi Secondary smoke-test 1`] = `
<div data-overlay-container="true">
  <button class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium opacity-disabled pointer-events-none [&amp;>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-secondary text-secondary-foreground data-[hover=true]:opacity-hover"
          data-disabled="true"
          data-loading="true"
          type="button"
          disabled
  >
    <div aria-label="Loading"
         class="relative inline-flex flex-col gap-2 items-center justify-center"
    >
      <div class="relative flex w-5 h-5">
        <i class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-l-transparent border-r-transparent border-2 border-b-current">
        </i>
        <i class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-l-transparent border-r-transparent border-2 border-b-current">
        </i>
      </div>
    </div>
    Loading
  </button>
</div>
`;
